### DISCUSSION TITLE
Monthly Rolling

### DISCUSSION THREAD

**Discussion by babameme** [pydata/xarray#5166] [Q&A]
Hi team,
I am trying to rolling by Monthly with Xarray DataArray
With Pandas Dataframe, i can use:
df.rolling("30D") or df.rolling("1M")
How can i do this with Xarray ?
Thanks

**Comment by fmaussion** [comment]
http://xarray.pydata.org/en/stable/generated/xarray.DataArray.rolling.html 

What did you try? What did not work out?

**Comment by babameme** [comment]
Yes, i follow that guidance.
I have try
da.rolling(dim = x).mean()
dim: One of DataArray da dimensions
x: int

That guide doesn't explicitly explain use cases of kwargs:
**window_kwargs (optional) – The keyword arguments form of dim. One of dim or window_kwargs must be provided.

**Comment by keewis** [comment]
`window_kwargs` is literally the keyword argument form of `dim` (it should have been named `window` to make this a little bit clearer). These are equivalent:
```python
da.rolling({"x": 3})  # dim
da.rolling(x=3)  # window_kwargs
```

however, you might actually be looking for `DataArray.resample`:
```python
da.resample({"time": "1M"})  # indexer form
da.resample(time="1M")  # indexer_kwargs form
```

Edit: see also #3330, which requests adding support for specifying the window size in the units of the dimension.
